plugin/pkg/admission/namespace/lifecycle/admission.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lifecycle
+
+import (
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/golang/glog"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/admission"
+	corev1 "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/pkg/api/v1"
+
+	kubeinformers "k8s.io/client-go/informers"
+	kubeclientset "k8s.io/client-go/kubernetes"
+
+	scadmission "github.com/kubernetes-incubator/service-catalog/pkg/apiserver/admission"
+)
+
+const (
+	// PluginName is name of admission plug-in
+	PluginName = "KubernetesNamespaceLifecycle"
+	// how long to wait for a missing namespace before re-checking the cache (and then doing a live lookup)
+	// this accomplishes two things:
+	// 1. It allows a watch-fed cache time to observe a namespace creation event
+	// 2. It allows time for a namespace creation to distribute to members of a storage cluster,
+	//    so the live lookup has a better chance of succeeding even if it isn't performed against the leader.
+	missingNamespaceWait = 50 * time.Millisecond
+)
+
+func init() {
+	admission.RegisterPlugin(PluginName, func(config io.Reader) (admission.Interface, error) {
+		return NewLifecycle()
+	})
+}
+
+// lifecycle is an implementation of admission.Interface.
+// It enforces life-cycle constraints around a Namespace depending on its Phase
+type lifecycle struct {
+	*admission.Handler
+	client          kubeclientset.Interface
+	namespaceLister corev1.NamespaceLister
+}
+
+type forceLiveLookupEntry struct {
+	expiry time.Time
+}
+
+var _ = scadmission.WantsKubeInformerFactory(&lifecycle{})
+var _ = scadmission.WantsKubeClientSet(&lifecycle{})
```

Comment:
We lose code readability. This is only 2 lines, but it's easy for someone to come into the codebase and ask what these lines do (I've seen it, and the questions add up). The answer is that this is kube convention as far as I can tell. This is a place to diverge from Kubernetes convention because it doesn't remove functionality, make it harder for us to integrate with the Kubernetes upstream packages, or introduce bugs.

As far as blocking this PR with my requested change, you're free to merge this without making this change, but I would like us all to start being more cognizant about changes like this in our PRs as we enter beta (which we are aiming to do in two weeks).